### ISSUE TITLE
Worker Wrapper Type Regression

### ISSUE THREAD

**Issue by n-ce** [Namide/extract-colors#56]
![Screenshot_20241015-201855_1](https://github.com/user-attachments/assets/a1e95738-122e-442b-a123-8239aa2e16f7)

Also if you don't mind looking at the parameters that I'm using, if they are  proper or not.

My OG theming vs extract-colors theming
![Screenshot_20241015-204506](https://github.com/user-attachments/assets/5bede80f-6b09-4dc4-9f8d-fb749f2434f1)
![Screenshot_20241015-204501](https://github.com/user-attachments/assets/f5e622de-efbf-4093-988c-b3cc9cdb751b)

I think its too saturated

**Comment by Namide** [comment]
Hello @n-ce,

Thank you for your issue.

## Type unknown

I've fix your first point in the last release `v4.1.1`: the worker return type.  
Can you confirm it the last release is better please?


## Color error

About your second point I've try with a downloaded image and the colors seems good:
![81ZnVYpTdhL _SY355_](https://github.com/user-attachments/assets/2f5d57ba-7d48-4cc6-acf6-48bdf6e3f474)

![image](https://github.com/user-attachments/assets/d4d828b9-d6bc-4b1a-bee6-ee437ad98e1f)

The options are the same than theses: [extract-colors.namide.com/demo](https://extract-colors.namide.com/demo).  
You can try by uploading the image directly on the website.

Can you share your image please?  
I will try with your image.

**Comment by n-ce** [comment]
Actually the thing is a color by itself is pretty useless especially in image color extraction.

I have to pass it through a lightener or darkener for using in a light dark/theme.

Now, in my program I'm using Average Color Algorithm which works for 4/5 cases. This is why I needed `extract-colors` for better accuracy of colors. 

But the problem is the lack of suitable color conversion. Maybe you can add that, it solves two problems
- Finding a suitable color in the colorArray
- Creating a color scheme out of the suitable color

**Comment by Namide** [comment]
For some time now I've been thinking about adding a feature to `extract-colors`:
I'd like a final step that classifies colors by “type” so as to be able to exploit them more easily.

I thought, for example, that `extract-colors` could return values such as these:
- main color
- secondary color
- main background color
- secondary background color
- most vivid color
- dullest color
- most used color
- …

But this implies returning empty values if no colors are found, because not all image types would match these results.
Moreover I don't yet know which color types would be relevant for the majority of cases.

In short, I don't know which options would be the most relevant and I'm still thinking about it before adding this feature.

But about creating a color scheme different from the original image, I hadn't thought of that. I wonder if this is beyond the scope of a color extractor.

**Comment by n-ce** [comment]
> But this implies returning empty values if no colors are found, because not all image types would match these results. 

Returning neutral values would be okay in these cases.

> Moreover I don't yet know which color types would be relevant for the majority of cases.

This is the exact problem I was facing, there is no single configuration which can lead to harmonious color schemes for most images.


> I wonder if this is beyond the scope of a color extractor.

You are right, it is beyond the scope of a color extractor. But ultimately it is your decision whether to add it or not and If you are ready to maintain the extra facilities in case you do.

closing this issue, because the Type issue is fixed.

**Comment by Namide** [comment]
I started working on this feature @n-ce: 
- [description](https://github.com/Namide/extract-colors/issues/58)
- [branch](https://github.com/Namide/extract-colors/tree/feature/classify)
